### PR TITLE
add display options to summary components

### PR DESCRIPTION
**Pull request by nkylstad**
## Description
Updated summary components with some display options to make it more readable with repeating groups:
- `hideBottomBorder` - to hide the blue dotted border between summary components.
- `useComponentGrid` - allows the summary component to use the layout components defined grid setup. Especially useful for summary groups with `largeGroup: true`.
- `hideChangeButton` - Useful when using summary components f.ex. in a custom confirm view, when it is not possible to navigate directly back to the form to change things.

In addition, updated some styling as discussed with @Febakke:
- Added a pale grey border around repeating group items in summary to make it more readable
- Swapped styling of label/data so that data is bold while label is normal.

Screenshot 2022-09-26 at 13 17 02

Apart from the small styling fixes, existing apps using summary components should not be affected.

## Related Issue(s)
- #461

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/altinn-app-frontend/src/components/summary/SummaryBoilerplate.tsx [hunk 1]
```diff
@@ -13,6 +14,7 @@ export interface SummaryBoilerplateProps {
   changeText: string;
   label: any;
   readOnlyComponent?: boolean;
+  display?: SummaryDisplayProperties;
```
Comment (olemartinorg) [inline src/altinn-app-frontend/src/components/summary/SummaryBoilerplate.tsx:17]: This should probably be added to `ILayoutCompSummary` instead, and this should extend from that interface. That would give you intellisense on `component.display...` after checking `if (component.type === 'Summary')` :smiley:

File: src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx [hunk 2]
```diff
@@ -14,13 +15,18 @@ import {
...
+  ILayoutComponent,
+  SummaryDisplayProperties,
+} from 'src/features/form/layout';
 import type { IComponentValidations, IRuntimeState } from 'src/types';
 
 export interface ISummaryComponent {
   id: string;
   componentRef?: string;
   pageRef?: string;
+  display?: SummaryDisplayProperties;
```
Comment (olemartinorg) [inline src/altinn-app-frontend/src/components/summary/SummaryComponent.tsx:29]: Same here. There are probably also some other options here not yet documented in the layout jsonschema..? :thinking:

**Review by olemartinorg — CHANGES_REQUESTED** [review]
:godmode:
I'll put up a 'request changes' because of the failing unit tests, otherwise this LGTM!

File: src/altinn-app-frontend/src/components/summary/SummaryBoilerplate.tsx (hunk 1 above)
Comment (nkylstad) [reply src/altinn-app-frontend/src/components/summary/SummaryBoilerplate.tsx]: Ah, great tip! Was able to get rid of several props that were already defined in `ILayoutCompSummary`. Will update jsonschema and make sure that all props are defined there.

**Review by olemartinorg — APPROVED** [review]

**Pull request merged**